hw11-cache/homework/src/main/java/ru/otus/HomeWork.java

```diff
@@ -0,0 +1,106 @@
+package ru.otus;
+
+import org.flywaydb.core.Flyway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.otus.cachehw.HwListener;
+import ru.otus.cachehw.MyCache;
+import ru.otus.core.repository.executor.DbExecutorImpl;
+import ru.otus.core.sessionmanager.TransactionRunner;
+import ru.otus.core.sessionmanager.TransactionRunnerJdbc;
+import ru.otus.crm.datasource.DriverManagerDataSource;
+import ru.otus.crm.model.Client;
+import ru.otus.crm.service.DBServiceClient;
+import ru.otus.crm.service.DbServiceClientImpl;
+import ru.otus.jdbc.mapper.*;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.WeakHashMap;
+
+public class HomeWork {
+    private static final String URL = "jdbc:postgresql://localhost:5430/demoDB";
+    private static final String USER = "usr";
+    private static final String PASSWORD = "pwd";
+
+    private static final Logger log = LoggerFactory.getLogger(HomeWork.class);
+
+    public static void main(String[] args) {
+
+        var dataSource = new DriverManagerDataSource(URL, USER, PASSWORD);
+        flywayMigrations(dataSource);
+        var transactionRunner = new TransactionRunnerJdbc(dataSource);
+        var dbExecutor = new DbExecutorImpl();
+
+        EntityClassMetaData entityClassMetaDataClient = new EntityClassMetaDataImpl<Client>(Client.class);
+        EntitySQLMetaData entitySQLMetaDataClient = new EntitySQLMetaDataImpl(entityClassMetaDataClient);
+        RowDataMapper rowDataMapperClient = new RowDataMapperImpl<Client>(entityClassMetaDataClient);
+        var dataTemplateClient = new DataTemplateJdbc<Client>(dbExecutor, entitySQLMetaDataClient, rowDataMapperClient);
+
+        actionsWithoutCache(transactionRunner, dataTemplateClient);
+        //actionsWithCache(transactionRunner, dataTemplateClient);
+
+    }
+
+    private static void flywayMigrations(DataSource dataSource) {
+        log.info("db migration started...");
+        var flyway = Flyway.configure()
+                .dataSource(dataSource)
+                .locations("classpath:/db/migration")
+                .load();
+        flyway.migrate();
+        log.info("db migration finished.");
+        log.info("***");
+    }
+
+    private static void actionsWithoutCache(TransactionRunner tr, DataTemplateJdbc dt) {
+        doAction(new DbServiceClientImpl(tr, dt, null));
```

Comment:
```suggestion
        doAction(new DbServiceClientImpl(tr, dt));
```

Лучше сделать отдельный конструктор без `MyCache`, чтобы не передавать `null`.

Reply:
исправлено